addon/serializers/json-api.js

```diff
@@ -153,10 +153,18 @@ const JSONAPISerializer = JSONSerializer.extend({
 
     if (Array.isArray(documentHash.included)) {
       let ret = new Array(documentHash.included.length);
+      let needsCompacting = false;
 
       for (let i = 0; i < documentHash.included.length; i++) {
         let included = documentHash.included[i];
-        ret[i] = this._normalizeResourceHelper(included);
+        let normalized = this._normalizeResourceHelper(included);
+        if (!normalized) {
+          needsCompacting = true;
+        }
+        ret[i] = normalized;
+      }
+      if (needsCompacting) {
+        ret = ret.filter(Boolean);
```

Comment:
I think we should just use push and an explicit check for not null vs preallocating length here.

Reply:
Sounds good @runspired. I was concerned about changing the performance characteristics of the happy path, but I agree that is clearer. I'll make the change now.

Reply:
✅